### PR TITLE
Switched email from gmail to sparkpost

### PR DESCRIPTION
**Pull request by Mantid-Matthew**
This PR switches the email client used to spark, it also updates the email message to contain a link to the errorreport service and filters out the names and email used by the system test so that email notifications are not sent for this.

### CODE REVIEW

File: web/services/models.py [hunk 1]
```diff
@@ -56,9 +56,13 @@ class RecoveryFiles(models.Model):
 
 
 def send_email_notification(sender, instance, signal, *args, **kwargs):
-    name = instance.user.name if instance.user else ''
-    email = instance.user.email if instance.user else ''
-    text_box = instance.textBox
+    name = instance.user.name if instance.user and \
```
Comment (martyngigg) [inline web/services/models.py]: Can we extract out the "magic" strings to module constants?

File: web/services/models.py [hunk 2]
```diff
@@ -56,9 +56,13 @@ class RecoveryFiles(models.Model):
...
+        instance.user.name != 'public_name' else ''
```
Comment (martyngigg) [inline web/services/models.py]: Do we want to assert on everything or just the minimal, obviously fake, email address?

File: web/settings.py [hunk 3]
```diff
@@ -148,8 +148,8 @@
     }
 
 EMAIL_USE_TLS = True
-EMAIL_HOST = 'smtp.gmail.com'
-EMAIL_HOST_USER = 'mantidproject@gmail.com'
+EMAIL_HOST = os.getenv('EMAIL_HOST', '')
```
Comment (martyngigg) [inline web/settings.py]: :+1: for being able to customize the email notification settings completely.

File: web/celery_app/tasks.py [hunk 4]
```diff
@@ -10,13 +10,15 @@ def send_notification_email(name, email, text_box):
...
+        ' who provided the following email {}. '
+        '\nThe following information was sent:\n\n {}'
+        '\n'
+        '\n https://errorreports.mantidproject.org/admin/'
     ).format(name, email, text_box)
 
     send_mail(subject,
               message,
-              'mantidproject@gmail.com',
+              'error-reports@mantidproject.org',
```
Comment (martyngigg) [inline web/celery_app/tasks.py]: Maybe we shouldn't hard code this either? Can we set it as a configurable variable too?

File: web/services/models.py (hunk 2 above)
Comment (Mantid-Matthew) [reply web/services/models.py]: There are some tests in the system test which don't provide an email address as all combinations of what may be provided are checked. Currently a notifiaction email is sent if the user, provides a name or an email or puts information in the text box so we need to check all three. 

Would it be better to only send a notification email if an email is provided.

File: README.md [hunk 5]
```diff
@@ -35,7 +35,8 @@ To start the services locally you will first need to create a `.env` file next t
         EMAIL_HOST_USER=SMTP_Injection
         EMAIL_PORT=587
         EMAIL_HOST_PASSWORD=<Api password need to retrieve from sparkpost>
-        ERROR_EMAIL=<email to recieve reports>   
+        ERROR_EMAIL=<email to recieve reports>
```
Comment (martyngigg) [inline README.md]: Could I be picky and we use the `EMAIL` prefix for all of the email related variables with something like
```
EMAIL_FROM_ADDRESS=
EMAIL_TO_ADDRESS=
```
for the last 2 variables?

**Pull request merged**